scilifelab/pm/core/deliver.py

```diff
@@ -126,6 +126,13 @@ def raw_data(self):
         # Extract the list of samples and runs associated with the project and sort them
         samples = sorted(s_con.get_samples(fc_id=self.pargs.flowcell, sample_prj=self.pargs.project), key=lambda k: (k.get('project_sample_name','NA'), k.get('flowcell','NA'), k.get('lane','NA')))
 
+        # Deliver only mentioned samples if any
+        if self.pargs.sample:
+            samples = [sample_obj for sample_obj in samples if sample_obj.get('project_sample_name','NA') == self.pargs.sample]
+            if len(samples) == 0:
```

Comment:
if not samples:

should work too, as empty lists evaluate to False, but it doesnt really matter